fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -871,14 +908,11 @@ private void checkInteractionModel(final UpdateRequest request) {
                 deleteQuads.addAll(op.getQuads());
             }
 
-            final Optional<String> resIxn = INTERACTION_MODELS.stream().filter(x -> hasType(x)).findFirst();
+            final Optional<String> resIxn = getResourceInteraction();
```

Comment:
You are checking for `resIxn.isPresent()` here as well as within `checkInteractionModel(triple, Optional<String>)`.
Can this be more efficient/consistent?